docs/content/advanced/backend/events/events-architecture.md

```diff
@@ -0,0 +1,85 @@
+The events system in Medusa is built on a publish/subscribe architecture. Events are published by the core when certain actions take place. Those events can be subscribed to, which means, a reactive action is performed asynchronously. It is useful in a variety of scenarios including but not limited to processing heavy jobs in the background and reacting to specific events in a custom project or plugin. 
+
+### Publishing and subscribing
+
+The `EventBusService` is responsible for publishing and processing events.
+
+> The current implementation of the `EventBusService` is powered by Redis. We will soon introduce an event bus module, allowing you to use any pub/sub provider. Therefore, the following documentation is based on the Redis implementation. Still, the general purpose and flows of the `EventBusService` will remain the same with the module implementation.
+
+The `EventBusService` exposes two methods in its public API for event processing; `emit` and `subscribe`. The former will take your event, add it to a Bull queue (powered by Redis) as a job, and process it asynchronously. The latter will attach a listener/subscriber to the specified event, which is run upon processing it. In this document, the terms event and job are used interchangeably.
```

Comment:
Suggestion: should we have a mention of the staged job as well at this point?

Reply:
I would like to keep out the staged job concept as much as possible, as it is not relevant to the consumer of the event bus service. They just need to be aware, that it happens under the hood.

Reply:
I ve seen that we mentioned it at a later point. So it might be enough. But i would argue that this is an important point in the customization. They will surely wrap their logic into a transaction and in that case the event might need to be emitted through the staged job.